src/ansys/platform/instancemanagement/configuration.py

```diff
@@ -0,0 +1,203 @@
+"""Configuration class module."""
+import json
+import logging
+import os
+import re
+from typing import Sequence, Tuple
+
+from ansys.platform.instancemanagement.exceptions import (
+    InvalidConfigurationError,
+    NotConfiguredError,
+)
+
+CONFIGURATION_PATH_ENVIRONMENT_VARIABLE = "ANSYS_PLATFORM_INSTANCEMANAGEMENT_CONFIG"
+
+logger = logging.getLogger(__name__)
+
+
+class Configuration:
+    """Configuration for the PIM client.
+
+    Raises:
+        InvalidConfigurationError: configuration file is not a well formatted json file
+        InvalidConfigurationError: version is not supported
+        InvalidConfigurationError: a key is missing in fthe configuration file
+
+    Returns:
+        Configuration: settings to configure the PIM client
+    """
+
+    _access_token: str
+    _headers: Sequence[Tuple[str, str]]
+    _tls: bool
+    _uri: str
+
+    @property
+    def access_token(self) -> str:
+        """Access token."""
+        return self._access_token
+
+    @property
+    def headers(self) -> Sequence[Tuple[str, str]]:
+        """Headers to add to the requests to PIM."""
+        return self._headers
+
+    @property
+    def tls(self) -> bool:
+        """Whether the connection to PIM requires encryption.
+
+        If ``True``, the ``access_token`` property is used to create a secure connection.
+
+        If ``False``, an unsecure connection is used.
+        """
+        return self._tls
+
+    @property
+    def uri(self) -> str:
+        """Uri of the PIM service."""
+        return self._uri
+
+    def __init__(
+        self, headers: Sequence[Tuple[str, str]], tls: bool, uri: str, access_token: str
+    ) -> None:
+        """Initialize the PIM configuration.
+
+        Parameters
+        ----------
+        tls
+
+        headers
+
+        """
+        self._access_token = access_token
+        self._headers = headers
+        self._tls = tls
+        self._uri = uri
+
+    @staticmethod
+    def from_file(config_path: str):
+        """Initialize the PyPIM configuration based on the configuration file.
+
+        Parameters
+        ----------
+        config_path : str
+            Path of the configuration file.
+
+        Returns
+        -------
+        Configuration
+            PyPIM configuration.
+
+        Raises
+        ------
+        InvalidConfigurationError
+            The configuration is not valid.
+        """
+        logger.debug("Initializing from %s", config_path)
+        with open(config_path, "r") as f:
+            try:
+                configuration = json.load(f)
+            except json.JSONDecodeError:
```

Comment:
when re-raising it's better to use this construct:

```python
except xx as e
    raise CustomException(...) from a
```

This makes the difference between an error cascade, where the "except" clause failed, and raising a specialized exception like this one